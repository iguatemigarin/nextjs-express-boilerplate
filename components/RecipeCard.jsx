import React from 'react'
import styled from 'styled-components'
import RecipeCard from '../ms_ui/src/components/RecipeCard'

const Wrapper = styled.div`
  padding: ${({ theme }) => theme.spacings.large};
`

const Content = styled.div`
  margin: 0 auto;
  max-width: 455px;
`

const StyledRecipeCard = () =>
  <Wrapper>
    <Content>
      <RecipeCard
        mealType='MEAT'
        title='The Recipe'
        subtitle='Delicious recipe'
        image={{url: "https://mscdn.freetls.fastly.net/media/recipes/15838/main_photos/medium/ID3070_web-3fe5bd2128f695215d913687d4e0e249.jpg"}} />
    </Content>
  </Wrapper>

export default StyledRecipeCard
