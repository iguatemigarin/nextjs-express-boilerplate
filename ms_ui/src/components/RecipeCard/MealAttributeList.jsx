import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const Attribute = styled.li`
  ${({ theme }) => `
    background-color: ${theme.tonals.t30}
    border-radius: ${theme.pill.borderRadius}
    color: ${theme.tonals.black}
    font: ${theme.fontStyle.tiny}
    margin-top: 0
    margin-bottom: ${theme.spacings.small}
    margin-left: 0
    margin-right: ${theme.brand === 'ms' ? theme.spacings.small : theme.spacings.medium}
    padding-top: ${theme.spacings.small}
    padding-bottom: ${theme.spacings.small}
    padding-right: ${theme.brand === 'ms' ? theme.spacings.small : theme.spacings.medium}
    padding-left: ${theme.brand === 'ms' ? theme.spacings.small : theme.spacings.medium}
  `}
  display: inline-block
  text-transform: uppercase
  letter-spacing: 1px
  line-height: 1.5
`;

const AttributesList = styled.ul`
  padding-left: 0
  margin: 0
`;

const MealAttributeList = ({ attributes }) => (
  <AttributesList>
    {attributes.map(({ key, name }) => <Attribute key={key}>{name}</Attribute>)}
  </AttributesList>
);

MealAttributeList.propTypes = {
  attributes: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string,
      name: PropTypes.string,
    }),
  ),
};

MealAttributeList.defaultProps = {
  attributes: [],
};

export default MealAttributeList;
