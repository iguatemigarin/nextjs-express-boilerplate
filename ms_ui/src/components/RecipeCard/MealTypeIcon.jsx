import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

const MealType = styled.div`
  ${({ theme, mealType }) => `
      background-image: url(${theme.mealTypeIcons[mealType]})}
      top: ${theme.spacings.mega}
      right: ${theme.spacings.mega}
      width: ${theme.iconSizes.large}
      height: ${theme.iconSizes.large}
  `}
  position: absolute
  border-radius: 50%
`;

const MealTypeIcon = ({ mealType }) => (
  <MealType mealType={mealType} />
);

MealTypeIcon.propTypes = {
  mealType: PropTypes.string.isRequired,
};

export default MealTypeIcon;
