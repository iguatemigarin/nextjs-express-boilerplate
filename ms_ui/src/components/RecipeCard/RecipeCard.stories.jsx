import { storiesOf } from '@storybook/react';
import {
  boolean,
  object,
  radios,
  select,
  text,
} from '@storybook/addon-knobs';
import React from 'react';
import styled from 'styled-components';
import RecipeCard from '.';
import { msTheme } from '../../themes/index';

const Wrapper = styled.div`
  ${({ theme }) => `
    display: flex;
    justify-content: center;
    padding: 40px 0;
    background: ${theme.tonals.t30};
  `}
`;

const Content = styled.div`
  max-width: 455px;
`;

storiesOf('RecipeCard', module)
  .addDecorator(story => (
    <Wrapper theme={msTheme}>
      <Content>
        {story()}
      </Content>
    </Wrapper>
  ))
  .addParameters({
    info: {
      text: 'Recipe card story',
      propTablesExclude: [Wrapper, Content],
    },
  })
  .add('with text', () => {
    const img1 = { url: 'https://mscdn.freetls.fastly.net/media/recipes/15838/main_photos/medium/ID3070_web-3fe5bd2128f695215d913687d4e0e249.jpg' };
    const img2 = { url: 'https://mscdn.freetls.fastly.net/media/recipes/17636/main_photos/medium/D127_Hero_2-ab0dc0eefada26fa0d09279ea60c2950.jpg' };

    const imageKnob = select('image', { img1, img2 }, img1);
    const titleKnob = text('title', 'Just Fry It');
    const subtitleKnob = text('subtitle', 'Subtitle It');
    const isClassicKnob = boolean('isClasic', false);
    const attributesKnob = object('attributes', [
      { key: 'kid_friendly', name: 'kinderfreundlich' },
      { key: 'healthy_choice', name: 'Fit & Gesund' },
      { key: 'poultry', name: 'Gefl√ºgel' },
      { key: 'no_gluten_ingredients', name: 'glutenarm' },
    ]);
    const mealTypeKnob = radios('mealType', { meat: 'MEAT', veggie: 'VEGGIE' }, 'MEAT');

    return (
      <RecipeCard
        image={imageKnob}
        title={titleKnob}
        subtitle={subtitleKnob}
        isClassic={isClassicKnob}
        attributes={attributesKnob}
        mealType={mealTypeKnob}
      />
    );
  });
