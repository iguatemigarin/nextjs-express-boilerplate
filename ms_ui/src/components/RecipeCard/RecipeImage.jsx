import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import MealTypeIcon from './MealTypeIcon';

const RecipeImageContainer = styled.div`
  position: relative
  height: auto
`;

const Image = styled.img`
  display: block
`;

const RecipeImage = ({ alt, mealType, src }) => (
  <RecipeImageContainer>
    <MealTypeIcon mealType={mealType} />
    <Image src={src} alt={alt} />
  </RecipeImageContainer>
);

RecipeImage.propTypes = {
  alt: PropTypes.string.isRequired,
  mealType: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired,
};

export default RecipeImage;
