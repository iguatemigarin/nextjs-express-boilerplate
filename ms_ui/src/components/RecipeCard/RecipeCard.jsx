// can we scope down what we require e.g. React.Component only
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import RecipeImage from './RecipeImage';
import MealAttributeList from './MealAttributeList';

const Card = styled.div`
  overflow: hidden;
  width: 100%;
  border-radius: ${props => props.theme.borderRadius.normal};
`;

const RecipeInfo = styled('div')`
  padding: ${props => props.theme.spacings.large}
  background-color: ${props => props.theme.colors.white}
`;

const RecipeTitle = styled('h2')`
  color: ${props => props.theme.colors.black}
  font: ${props => props.theme.fontStyle.h5}
`;

const RecipeSubtitle = styled('div')`
  font: ${props => props.theme.fontStyle.body}
  margin: 0 0 15px
  color: ${props => props.theme.tonals.t10}
`;

class RecipeCard extends React.Component {
  constructor(...args) {
    super(...args);
    this.onCardClick = this.onCardClick.bind(this);
  }

  onCardClick() {
    const { onCardClick } = this.props;
    onCardClick();
  }

  render() {
    const {
      image,
      mealType,
      title,
      subtitle,
      attributes,
    } = this.props;

    return (
      <Card onClick={this.onCardClick}>
        <RecipeImage src={image.url} mealType={mealType} alt={title} />
        <RecipeInfo>
          <RecipeTitle>{title}</RecipeTitle>
          <RecipeSubtitle>{subtitle}</RecipeSubtitle>
          <MealAttributeList attributes={attributes} />
        </RecipeInfo>
      </Card>
    );
  }
}

// Why pass an image object? This could just be imageUrl
RecipeCard.propTypes = {
  onCardClick: PropTypes.func,
  image: PropTypes.shape({ url: PropTypes.string }).isRequired,
  mealType: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  attributes: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string,
      name: PropTypes.string,
    }),
  ),
};

RecipeCard.defaultProps = {
  onCardClick: () => {},
  attributes: [],
};

export default RecipeCard;
