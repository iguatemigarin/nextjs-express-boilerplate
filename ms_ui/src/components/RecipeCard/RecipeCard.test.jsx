import RecipeCard from './RecipeCard';

const onCardClickMock = jest.fn();
const exampleCard = (
  <RecipeCard
    image={{ url: 'https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg' }}
    title="Just Fry It MS"
    subtitle="Subtitle for MS"
    isClassic={false}
    attributes={[
      { key: 'kid_friendly', name: 'kinderfreundlich' },
      { key: 'healthy_choice', name: 'Fit & Gesund' },
      { key: 'poultry', name: 'GeflÃ¼gel' },
      { key: 'no_gluten_ingredients', name: 'glutenarm' },
    ]}
    mealType="MEAT"
    onCardClick={onCardClickMock}
  />
);

describe('RecipeCard', () => {
  it('renders correctly with ms theme', () => {
    expect(toJSON(mountWithTheme(exampleCard, msTheme))).toMatchSnapshot();
  });

  it('renders correctly with dn theme', () => {
    expect(toJSON(mountWithTheme(exampleCard, dnTheme))).toMatchSnapshot();
  });

  it('it calls the correct event handler when clicking the card', () => {
    shallow(exampleCard).simulate('click');
    expect(onCardClickMock).toHaveBeenCalledTimes(1);
  });
});
