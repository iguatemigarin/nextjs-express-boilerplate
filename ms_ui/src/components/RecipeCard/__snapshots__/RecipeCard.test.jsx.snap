// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard renders correctly with dn theme 1`] = `
.c2 {
  background-image: url(/images/dn/meat.svg);
  top: 20px;
  right: 20px;
  width: 37px;
  height: 37px;
  position: absolute;
  border-radius: 50%;
}

.c1 {
  position: relative;
  height: auto;
}

.c3 {
  display: block;
}

.c8 {
  background-color: #f4f2f9;
  border-radius: 100px;
  color: undefined;
  font: 10px/1.5 "Galano Grotesque Medium",sans-serif;
  margin-top: 0;
  margin-bottom: 5px;
  margin-left: 0;
  margin-right: 10px;
  padding-top: 5px;
  padding-bottom: 5px;
  padding-right: 10px;
  padding-left: 10px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5;
}

.c7 {
  padding-left: 0;
  margin: 0;
}

.c0 {
  overflow: hidden;
  width: 100%;
  border-radius: 8px;
}

.c4 {
  padding: 15px;
  background-color: #ffffff;
}

.c5 {
  color: #3a3a3a;
  font: 24px/0 "Galano Grotesque Medium",sans-serif;
}

.c6 {
  font: 16px/1.5 "Galano Grotesque Regular",sans-serif;
  margin: 0 0 15px;
  color: #747475;
}

<RecipeCard
  attributes={
    Array [
      Object {
        "key": "kid_friendly",
        "name": "kinderfreundlich",
      },
      Object {
        "key": "healthy_choice",
        "name": "Fit & Gesund",
      },
      Object {
        "key": "poultry",
        "name": "Geflügel",
      },
      Object {
        "key": "no_gluten_ingredients",
        "name": "glutenarm",
      },
    ]
  }
  image={
    Object {
      "url": "https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg",
    }
  }
  isClassic={false}
  mealType="MEAT"
  onCardClick={[MockFunction]}
  subtitle="Subtitle for MS"
  title="Just Fry It MS"
>
  <styled.div
    onClick={[Function]}
  >
    <div
      className="c0"
      onClick={[Function]}
    >
      <RecipeImage
        alt="Just Fry It MS"
        mealType="MEAT"
        src="https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg"
      >
        <styled.div>
          <div
            className="c1"
          >
            <MealTypeIcon
              mealType="MEAT"
            >
              <styled.div
                mealType="MEAT"
              >
                <div
                  className="c2"
                />
              </styled.div>
            </MealTypeIcon>
            <styled.img
              alt="Just Fry It MS"
              src="https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg"
            >
              <img
                alt="Just Fry It MS"
                className="c3"
                src="https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg"
              />
            </styled.img>
          </div>
        </styled.div>
      </RecipeImage>
      <styled.div>
        <div
          className="c4"
        >
          <styled.h2>
            <h2
              className="c5"
            >
              Just Fry It MS
            </h2>
          </styled.h2>
          <styled.div>
            <div
              className="c6"
            >
              Subtitle for MS
            </div>
          </styled.div>
          <MealAttributeList
            attributes={
              Array [
                Object {
                  "key": "kid_friendly",
                  "name": "kinderfreundlich",
                },
                Object {
                  "key": "healthy_choice",
                  "name": "Fit & Gesund",
                },
                Object {
                  "key": "poultry",
                  "name": "Geflügel",
                },
                Object {
                  "key": "no_gluten_ingredients",
                  "name": "glutenarm",
                },
              ]
            }
          >
            <styled.ul>
              <ul
                className="c7"
              >
                <styled.li
                  key="kid_friendly"
                >
                  <li
                    className="c8"
                  >
                    kinderfreundlich
                  </li>
                </styled.li>
                <styled.li
                  key="healthy_choice"
                >
                  <li
                    className="c8"
                  >
                    Fit & Gesund
                  </li>
                </styled.li>
                <styled.li
                  key="poultry"
                >
                  <li
                    className="c8"
                  >
                    Geflügel
                  </li>
                </styled.li>
                <styled.li
                  key="no_gluten_ingredients"
                >
                  <li
                    className="c8"
                  >
                    glutenarm
                  </li>
                </styled.li>
              </ul>
            </styled.ul>
          </MealAttributeList>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RecipeCard>
`;

exports[`RecipeCard renders correctly with ms theme 1`] = `
.c2 {
  background-image: url(/images/ms/meat.svg);
  top: 15px;
  right: 15px;
  width: 37px;
  height: 37px;
  position: absolute;
  border-radius: 50%;
}

.c1 {
  position: relative;
  height: auto;
}

.c3 {
  display: block;
}

.c8 {
  background-color: #f6f6f6;
  border-radius: none;
  color: undefined;
  font: 10px/1.5 "Avenir Demi",sans-serif;
  margin-top: 0;
  margin-bottom: 5px;
  margin-left: 0;
  margin-right: 5px;
  padding-top: 5px;
  padding-bottom: 5px;
  padding-right: 5px;
  padding-left: 5px;
  display: inline-block;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 1.5;
}

.c7 {
  padding-left: 0;
  margin: 0;
}

.c0 {
  overflow: hidden;
  width: 100%;
  border-radius: 8px;
}

.c4 {
  padding: 15px;
  background-color: #ffffff;
}

.c5 {
  color: #2d2926;
  font: 20px/0 "Avenir Medium",sans-serif;
}

.c6 {
  font: 16px/1.5 "Avenir Regular",sans-serif;
  margin: 0 0 15px;
  color: #413b38;
}

<RecipeCard
  attributes={
    Array [
      Object {
        "key": "kid_friendly",
        "name": "kinderfreundlich",
      },
      Object {
        "key": "healthy_choice",
        "name": "Fit & Gesund",
      },
      Object {
        "key": "poultry",
        "name": "Geflügel",
      },
      Object {
        "key": "no_gluten_ingredients",
        "name": "glutenarm",
      },
    ]
  }
  image={
    Object {
      "url": "https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg",
    }
  }
  isClassic={false}
  mealType="MEAT"
  onCardClick={[MockFunction]}
  subtitle="Subtitle for MS"
  title="Just Fry It MS"
>
  <styled.div
    onClick={[Function]}
  >
    <div
      className="c0"
      onClick={[Function]}
    >
      <RecipeImage
        alt="Just Fry It MS"
        mealType="MEAT"
        src="https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg"
      >
        <styled.div>
          <div
            className="c1"
          >
            <MealTypeIcon
              mealType="MEAT"
            >
              <styled.div
                mealType="MEAT"
              >
                <div
                  className="c2"
                />
              </styled.div>
            </MealTypeIcon>
            <styled.img
              alt="Just Fry It MS"
              src="https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg"
            >
              <img
                alt="Just Fry It MS"
                className="c3"
                src="https://mscdn.freetls.fastly.net/media/recipes/12295/main_photos/medium/ID0723_web-cc638b0227eefd29aab6c0e71a930475.jpg"
              />
            </styled.img>
          </div>
        </styled.div>
      </RecipeImage>
      <styled.div>
        <div
          className="c4"
        >
          <styled.h2>
            <h2
              className="c5"
            >
              Just Fry It MS
            </h2>
          </styled.h2>
          <styled.div>
            <div
              className="c6"
            >
              Subtitle for MS
            </div>
          </styled.div>
          <MealAttributeList
            attributes={
              Array [
                Object {
                  "key": "kid_friendly",
                  "name": "kinderfreundlich",
                },
                Object {
                  "key": "healthy_choice",
                  "name": "Fit & Gesund",
                },
                Object {
                  "key": "poultry",
                  "name": "Geflügel",
                },
                Object {
                  "key": "no_gluten_ingredients",
                  "name": "glutenarm",
                },
              ]
            }
          >
            <styled.ul>
              <ul
                className="c7"
              >
                <styled.li
                  key="kid_friendly"
                >
                  <li
                    className="c8"
                  >
                    kinderfreundlich
                  </li>
                </styled.li>
                <styled.li
                  key="healthy_choice"
                >
                  <li
                    className="c8"
                  >
                    Fit & Gesund
                  </li>
                </styled.li>
                <styled.li
                  key="poultry"
                >
                  <li
                    className="c8"
                  >
                    Geflügel
                  </li>
                </styled.li>
                <styled.li
                  key="no_gluten_ingredients"
                >
                  <li
                    className="c8"
                  >
                    glutenarm
                  </li>
                </styled.li>
              </ul>
            </styled.ul>
          </MealAttributeList>
        </div>
      </styled.div>
    </div>
  </styled.div>
</RecipeCard>
`;
